crates/node/service/src/service/core.rs

```diff
@@ -96,82 +79,60 @@ pub trait RollupNodeService {
             Error: Display,
             InboundData = SupervisorActorContext,
-            State = Self::SupervisorExt,
             OutboundData = SupervisorOutboundData,
         >;
 
     /// The type of runtime actor to use for the service.
-    type RuntimeActor: NodeActor<
-            Error: Display,
-            InboundData = RuntimeContext,
-            State = RuntimeState,
-            OutboundData = RuntimeOutboundData,
-        >;
+    type RuntimeActor: NodeActor<Error: Display, InboundData = RuntimeContext, OutboundData = RuntimeOutboundData>;
 
     /// The type of rpc actor to use for the service.
-    type RpcActor: NodeActor<Error: Display, InboundData = RpcContext, State = RpcLauncher, OutboundData = ()>;
+    type RpcActor: NodeActor<Error: Display, InboundData = RpcContext, OutboundData = (), Builder = RpcBuilder>;
 
     /// The type of error for the service's entrypoint.
     type Error: From<RpcLauncherError>
         + From<jsonrpsee::server::RegisterMethodError>
         + std::fmt::Debug;
 
-    /// Returns the [`NodeMode`] of the service.
+    /// The mode of operation for the node.
     fn mode(&self) -> NodeMode;
 
-    /// Returns a reference to the rollup node's [`RollupConfig`].
-    fn config(&self) -> Arc<RollupConfig>;
+    /// Returns a DA watcher builder for the node.
+    fn da_watcher_builder(&self) -> <Self::DataAvailabilityWatcher as NodeActor>::Builder;
 
-    /// Returns the [`RootProvider`] for the L1 chain.
-    fn l1_provider(&self) -> RootProvider;
+    /// Returns a derivation builder for the node.
+    fn derivation_builder(&self) -> <Self::DerivationActor as NodeActor>::Builder;
 
-    /// Creates a new instance of the [`Pipeline`] and initializes it. Returns the starting L2
-    /// forkchoice state and the initialized derivation pipeline.
-    async fn init_derivation(&self) -> Result<Self::DerivationPipeline, Self::Error>;
+    /// Creates a network builder and the [`NetworkRpc`] for the node.
+    fn network_builder(&self) -> (<Self::NetworkActor as NodeActor>::Builder, NetworkRpc);
 
-    /// Creates a new instance of the [`Network`].
-    async fn init_network(&self) -> Result<(Network, NetworkRpc), Self::Error>;
+    /// Returns a runtime builder for the node.
+    fn runtime_builder(&self) -> Option<<Self::RuntimeActor as NodeActor>::Builder>;
+
+    /// Returns an engine builder for the node.
+    fn engine_builder(&self) -> <Self::EngineActor as NodeActor>::Builder;
 
     /// Creates a new [`Self::SupervisorExt`] to be used in the supervisor rpc actor.
     async fn supervisor_ext(&self) -> Option<Self::SupervisorExt>;
 
-    /// Returns the [`RuntimeState`] for the node.
-    fn runtime(&self) -> Option<&RuntimeState>;
-
-    /// Returns the [`EngineLauncher`]
-    fn engine(&self) -> EngineLauncher;
-
-    /// Returns the [`RpcLauncher`] for the node.
-    fn rpc(&self) -> RpcLauncher;
+    /// Returns the [`RpcBuilder`] for the node.
+    fn rpc(&self) -> RpcBuilder;
```

Comment:
We suffix all the other builder methods with `_builder`, let's add that here for consistency.